model/private_subnet.rb

```diff
@@ -142,8 +142,15 @@ def create_tunnels(nics, src_nic)
     end
   end
 
-  def find_all_connected_nics(excluded_private_subnet_ids = [])
-    nics + connected_subnets.select { |subnet| !excluded_private_subnet_ids.include?(subnet.id) }.flat_map { it.find_all_connected_nics(excluded_private_subnet_ids + [id]) }.uniq
```

Comment:
@furkansahin IIRC the general idea we had was that connected subnets would be bilateral only, to avoid tricky questions about routing via a connected subnet. That being the case, why is there a recursive routine in here?

Reply:
The recursion is used to resolve the relationship from nic a to nic c. Since the tunnels are created bidirectional, if we rekey the tunnels in between nic a and b, we want to also rekey all of the tunnels touching nic b which include nic c - nic b tunnels. This is needed because the rekey nic tunnel prog drops all of the state objects that are not part of the current rekeying operations. So, if we don't include the tunnel between nic b and nic c into that rekeying batch, that connection is dropped as part of the rekeying of the tunnel nic A-B.